google-ads-examples/src/main/java/com/google/ads/googleads/examples/extensions/AddGoogleMyBusinessLocationExtensions.java

```diff
@@ -0,0 +1,250 @@
+      // 1. Set up the FeedAttributes on the feed.
+      // 2. Set up a FeedMapping that associates the FeedAttributes of the feed
+      // with the placeholder fields of the LOCATION placeholder type.
+      MutateFeedsResponse response =
+          feedServiceClient.mutateFeeds(Long.toString(customerId), ImmutableList.of(operation));
+      String gmbFeedResourceName = response.getResults(0).getResourceName();
+      System.out.printf("GMB feed created with resource name: %s%n", gmbFeedResourceName);
+
+      // Adds a CustomerFeed that associates the feed with this customer for
+      // the LOCATION placeholder type.
+      CustomerFeed customerFeed =
+          CustomerFeed.newBuilder()
+              .setFeed(StringValue.of(gmbFeedResourceName))
+              .addPlaceholderTypes(PlaceholderType.LOCATION)
+              // Creates a matching function that will always evaluate to true.
+              .setMatchingFunction(
+                  MatchingFunction.newBuilder()
+                      .addLeftOperands(
+                          Operand.newBuilder()
+                              .setConstantOperand(
+                                  ConstantOperand.newBuilder()
+                                      .setBooleanValue(BoolValue.of(true))
+                                      .build())
+                              .build())
+                      .setFunctionString(StringValue.of("IDENTITY(true)"))
+                      .setOperator(MatchingFunctionOperator.IDENTITY)
+                      .build())
+              .build();
+
+      CustomerFeedOperation customerFeedOperation =
+          CustomerFeedOperation.newBuilder().setCreate(customerFeed).build();
+
+      try (CustomerFeedServiceClient customerFeedServiceClient =
+          googleAdsClient.getLatestVersion().createCustomerFeedServiceClient()) {
+
+        // After the completion of the Feed ADD operation above the added feed will not be available
+        // for usage in a CustomerFeed until the sync between the Google Ads and GMB accounts
+        // completes. The loop below will retry adding the CustomerFeed up to ten times with an
+        // exponential back-off policy.
+        String addedCustomerFeed = null;
+        int numberOfAttempts = 0;
+        do {
+          numberOfAttempts++;
+          try {
+            MutateCustomerFeedsResponse customerFeedsResponse =
+                customerFeedServiceClient.mutateCustomerFeeds(
+                    Long.toString(customerId), ImmutableList.of(customerFeedOperation));
+            addedCustomerFeed = customerFeedsResponse.getResults(0).getResourceName();
+            System.out.printf("Customer feed created with resource name: %s%n", addedCustomerFeed);
+          } catch (GoogleAdsException gae) {
+            // Waits using exponential backoff policy.
+            long sleepSeconds = (long) Math.scalb(5, numberOfAttempts);
```

Comment:
Should we add an upper bound on sleepSeconds? If MAX_CUSTOMER_FEED_ADD_ATTEMPTS is set too high, this sleep will become grow too large.